activityical.php

```diff
@@ -397,3 +397,26 @@ function activityical_civicrm_tokenValues(&$values, $cids, $job = NULL, $tokens
     }
   }
 }
+
+/**
+ * Implements hook_civicrm_merge().
+ */
+function activityical_civicrm_merge($type, &$data, $mainId = NULL, $otherId = NULL, $tables = NULL) {
+  if ($type == 'sqls') {
+    $deleteIcalContactInCache = civicrm_api3('ActivityicalCache', 'get', [
```

Comment:
It's correct that we want to delete the activityicalcache records for $mainId.

However, this code is doing the deletion with the API immediately, which is not ideal.

Instead we should _add an sql statement to $data that would make that deletion_. This way, civicrm will include that sql statement in the final series of changes that are made all at once (i.e., in an atomic mysql transaction) at the very end of the merge process.

@lucky091588 can you make that change?